polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -508,11 +508,8 @@ async fn distribute_collation<Context>(
 		state.collation_result_senders.insert(candidate_hash, result_sender);
 	}
 
-	let parent_head_data = if elastic_scaling {
-		ParentHeadData::WithData { hash: parent_head_data_hash, head_data: parent_head_data }
-	} else {
-		ParentHeadData::OnlyHash(parent_head_data_hash)
-	};
+	let parent_head_data =
```

Comment:
In `send_collation(` we should also be able to get rid of the match since all validators now support `CollationWithParentHeadData`.

Reply:
I wouldn't define a new data type just for this purpose, just pass both values to the send_collation

Reply:
actually, we don't even use the hash here

Reply:
Ok I think I've made sensible changes now.